doc/spec/support-assessment.md

```diff
@@ -0,0 +1,237 @@
+# Support Assessment Dashboard: Infrastructure
+
+## Problem Description
+
+The support assessment dashboard is a feature giving end users insight
+into how well their cluster conforms to a support envelope based on
+continuously measuring and evaluating system state. This document
+proposes infrastructure to effectively do this.
+
+We can roughly divide the problem into:
+1. Data sources measuring system state
+2. Data gathering, preparation, and joining multiple sources
+3. Rule evaluation
+4. User interface (alerts, notifications, dashboard)
+
+The document focuses on 2. 3. while touching 1.
+
+## Challenges
+
+- When gathering and joining data we must take into account that
+  keeping a copy of all state in memory won't scale
+- We must join data from diverse sources like the manger, Prometheus,
+  cluster maps, Aquarium, Ceph orchestrator. All having different schema.
+- Rules are subject to configuration and change. Changes here must be easy.
+
+## Proposed Change
+
+Use Prometheus to gather, prepare, join and evaluate rules on numeric
+metrics. Rules are Prometheus alerts, written in PromQL and tested
+with the [`promtool`](https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules)
+
+Begin with readily-available metrics from Prometheus node exporter and
+Ceph manager's Prometheus exporter. Implement custom Prometheus
+exporters where necessary. Abstract complex queries behind simple
+numeric metrics where appropriate and necessary due to the lack of a
+string metric type. Example: Metric `outdated_client 42`
+instead of metrics `client{version="23.42"} 42`.
```

Comment:
This is one such example, since it'd be useful to be able to report which clients run which versions; it could be fixed by tracking this as a BCD number (so we can have our own 2K22 bug at some point), or some other scheme that doesn't lose information.

Reply:
Do I grok this correctly, encode the version number as a gauge metric?

Something like this?
`client{type="cephfs", name="somehorst.example.com",pid="2342",implementation="libcephfs"} BCD_ENCODED_VERSION_NUMBER`